lib/sycamore/sycamore/transforms/extract_entity.py

```diff
@@ -84,27 +81,31 @@ def __init__(
         self._field = field
 
     @timetrace("OaExtract")
-    def extract_entity(self, document: Document) -> Document:
+    def extract_entity(self, document: Document, context: Optional[Context] = None) -> Document:
+        if self._llm is None and context:
+            self._llm = context.llm
+        assert self._llm is not None, "OpenAIEntityExtractor requires an LLM"
+
         if self._use_elements:
             if self._prompt_template:
-                entities = self._handle_few_shot_prompting(document)
+                entities = self._handle_few_shot_prompting(document, self._llm)
```

Comment:
why pass in self._llm to a method on self?

Reply:
Since the llm can now be set in __init__ and execute(..), it was a way to avoid having each llm access in the private function (_handle_xyz) be wrapped around a assert.self._llm is not None

Don't have strong opinions, this is less code